addons/binding/org.openhab.binding.elerotransmitterstick/src/main/java/org/openhab/binding/elerotransmitterstick/internal/discovery/EleroChannelDiscoveryService.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.elerotransmitterstick.internal.discovery;
+
+import static org.openhab.binding.elerotransmitterstick.internal.EleroTransmitterStickBindingConstants.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceCallback;
+import org.eclipse.smarthome.config.discovery.ExtendedDiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.elerotransmitterstick.internal.handler.EleroTransmitterStickHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EleroChannelDiscoveryService} is responsible for discovery of elero channels from an Elero Transmitter
+ * Stick.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class EleroChannelDiscoveryService extends AbstractDiscoveryService implements ExtendedDiscoveryService {
```

Comment:
`ExtendedDiscoveryService` is deprecated, please remove it

Reply:
Done.